### PR TITLE
Add test for AsyncCall failure in application Interceptor

### PR DESCRIPTION
**Pull request by NightlyNexus**
A failing test for https://github.com/square/okhttp/issues/1801
I am not sure this is the right place for the test, though.

### CODE REVIEW

File: okhttp-tests/src/test/java/com/squareup/okhttp/DispatcherTest.java [hunk 1]
```diff
@@ -24,6 +25,17 @@
     dispatcher.setMaxRequestsPerHost(10);
   }
 
+  @Test public void asyncCallEngineInitialized () {
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/DispatcherTest.java]: nit: space before `()`

File: okhttp-tests/src/test/java/com/squareup/okhttp/DispatcherTest.java [hunk 2]
```diff
@@ -24,6 +25,17 @@
...
+    final OkHttpClient client = new OkHttpClient();
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/DispatcherTest.java]: nit: we don't `final` on locals unless absolutely necessary

**Comment by swankjesse** [comment]
Probably `CallTest` is the best place for it. It's where all the tests go.

**Comment by NightlyNexus** [comment]
Closing, see: https://github.com/square/okhttp/pull/1870

**Pull request closed without merging**